velox/examples/SimpleFunctions.cpp

```diff
@@ -474,9 +476,106 @@ void register8() {
       std::shared_ptr<UserDefinedObject>>({"my_opaque_func"});
 }
 
-template <typename T>
+// Define a toy function that doubles the nested input parameters provided as an
+// Array, Map, or Row type.
+//
+// For nested complex types, add_item() is the most common API to access the
+// writer for the inner type. When adding a null entry is desired, add_null()
+// for Arrays, Maps and set_null_at<index>() for Row types can be used.
+template <typename TExecParams>
+struct MyNestedComplexTimesTwoFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExecParams);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Array<Map<int64_t, double>>>& result,
+      const arg_type<Array<Map<int64_t, double>>>& inputArrayOfMap) {
+    result.reserve(inputArrayOfMap.size());
+    for (const auto& innerMap : inputArrayOfMap) {
+      if (!innerMap.has_value()) {
```

Comment:
just in case, please comment the body as well. This is not "real" code, so the more we walk the reader through what is going on, the better.